ingestion/src/metadata/orm_profiler/interfaces/sqa_profiler_interface.py

```diff
@@ -132,7 +133,8 @@ def compute_metrics_in_thread(
         logger.debug(
             f"Running profiler for {table.__tablename__} on thread {threading.current_thread()}"
         )
-        session = self._session_factory()
+        Session = self.session_factory
+        session = Session()
```

Comment:
should we use `with Session() as session`  here to automatically handle open and close?

Reply:
@pmbrull basically we use a `scoped_session` which creates a thread safe session. `Session()` reffectively returns a `Session` object. The only difference with `sqa.Session()` is that 1 `Session` object will be created by local thread -- hence won't be shared across threads.

Reply:
This will be automatically garabage collected (if I understand well) once the application thread ends -- more info [here](https://docs.sqlalchemy.org/en/14/orm/contextual.html#thread-local-scope). I can actually see a new session object instance is created for each table that is processed.

> one advantage of the threading.local() object is that if the application thread itself ends, the “storage” for that thread is also garbage collected. So it is in fact “safe” to use thread local scope with an application that spawns and tears down threads, without the need to call [scoped_session.remove()](https://docs.sqlalchemy.org/en/14/orm/contextual.html#sqlalchemy.orm.scoping.scoped_session.remove).